flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2465,9 +2439,9 @@ struct FirEndOpConversion : public FIROpConversion<fir::FirEndOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::FirEndOp op, OperandTy operands,
+  matchAndRewrite(fir::FirEndOp firEnd, OperandTy operands,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    rewriter.replaceOp(op, {});
+    TODO(firEnd.getLoc(), "fir.end codegen");
```

Comment:
I think I get why this was removed, but `fir.end` is a nop.

Reply:
Diverge. Not sure `replaceOp` is the preferred way to erase this op.

Reply:
Using eraseOp to erase.